src/tests/test_dynamic_api/test_views.py

```diff
@@ -617,3 +617,48 @@ def test_detail_expand_true_non_temporal_many_to_many_to_temporal(
             "http://testserver/v1/gebieden/buurten/03630000000078/",
         ]
         assert response.data["_embedded"]["buurten"]["id"] == "03630000000078.2"
+
+    def test_independence_of_m2m_through_id_field(
+        self,
+        api_client,
+        reloadrouter,
+        buurten_model,
+        buurten_data,
+        woningbouwplan_model,
+        woningbouwplannen_data,
+    ):
+        """Prove that the many-to-many relation from a non-temporal to temporal dataset
+        works without using the 'id' column in the though table.
+        """
+        cursor = connection.cursor()
+        cursor.execute(
+            """SELECT column_name FROM information_schema.columns
+            WHERE table_schema = 'public'
+            AND table_name = 'woningbouwplannen_woningbouwplan_buurten';
+            """
+        )
+        column_names_before = set([column_name for (column_name,) in cursor.fetchall()])
+        assert "id" in column_names_before
+        """ renaming the 'id' column to 'wbw_rel_woningbouwplan_buurt_id'
+        to mimick the woningbouwplannen dataset"""
+        cursor.execute(
+            """ALTER TABLE woningbouwplannen_woningbouwplan_buurten
+            RENAME COLUMN id TO wbw_rel_woningbouwplan_buurt_id;
+            """
+        )
+        cursor.execute(
+            """SELECT column_name FROM information_schema.columns
+            WHERE table_schema = 'public'
+            AND table_name = 'woningbouwplannen_woningbouwplan_buurten';
+            """
+        )
+        column_names_after = set([column_name for (column_name,) in cursor.fetchall()])
+        assert "wbw_rel_woningbouwplan_buurt_id" in column_names_after
+        assert "id" not in column_names_after
+        url = reverse("dynamic_api:woningbouwplannen-woningbouwplan-detail", args=[1])
+        url = f"{url}?_expand=true"
+        response = api_client.get(url)
+        # check that "buurten" still contains the correct list
+        assert response.data["buurten"] == [
```

Comment:
Interessante test, maar deze PR is over de impliciete `ORDER` op queries en niet over de `ID` velden. Of mis ik iets?

Reply:
Het oorspronkelijke probleem was een afhankelijkheid van 'id' in de tussentabel. Dit kwam door een impliciete order op 'id'. Deze PR verwijdert de ORDER BY. De test checked of het einddoel behaald is: niet meer afhankelijk van 'id' in de tussentabel, door de naam van die kolom te veranderen in iets anders.

Reply:
de PR gaat dus eigenlijk niet over ordering, maar over afhankelijkheid van een 'id' veld.

Reply:
Duidelijk, thnx.